src/core/operations/Enigma.mjs

```diff
@@ -0,0 +1,199 @@
+        this.infoURL = "https://wikipedia.org/wiki/Enigma_machine";
+        this.inputType = "string";
+        this.outputType = "string";
+        this.args = [
+            {
+                name: "1st (right-hand) rotor",
+                type: "editableOption",
+                value: Enigma.ROTORS,
+                // Default config is the rotors I-III *left to right*
+                defaultIndex: 2
+            },
+            {
+                name: "1st rotor ring setting",
+                type: "option",
+                value: Enigma.LETTERS
+            },
+            {
+                name: "1st rotor initial value",
+                type: "option",
+                value: Enigma.LETTERS
+            },
+            {
+                name: "2nd rotor",
+                type: "editableOption",
+                value: Enigma.ROTORS,
+                defaultIndex: 1
+            },
+            {
+                name: "2nd rotor ring setting",
+                type: "option",
+                value: Enigma.LETTERS
+            },
+            {
+                name: "2nd rotor initial value",
+                type: "option",
+                value: Enigma.LETTERS
+            },
+            {
+                name: "3rd rotor",
+                type: "editableOption",
+                value: Enigma.ROTORS,
+                defaultIndex: 0
+            },
+            {
+                name: "3rd rotor ring setting",
+                type: "option",
+                value: Enigma.LETTERS
+            },
+            {
+                name: "3rd rotor initial value",
+                type: "option",
+                value: Enigma.LETTERS
+            },
+            {
+                name: "4th rotor",
+                type: "editableOption",
+                value: Enigma.ROTORS_OPTIONAL,
+                defaultIndex: 10
+            },
+            {
+                name: "4th rotor initial value",
+                type: "option",
+                value: Enigma.LETTERS
+            },
+            {
+                name: "Reflector",
+                type: "editableOption",
+                value: Enigma.REFLECTORS
+            },
+            {
+                name: "Plugboard",
+                type: "string",
+                value: ""
+            },
+            {
+                name: "Drop non-alphabet chars",
```

Comment:
Consider renaming this to "Strict output". You could add an explanation of exactly what this does in the description. You can also add a `hint: "Hint goes here"` property to this dictionary which triggers help text when you focus the checkbox, but I haven't really got the CSS sorted out yet, so it will likely not display correctly.